pkg/loki/modules.go

```diff
@@ -205,6 +205,12 @@ func (t *Loki) initDistributor() (services.Service, error) {
 		logproto.RegisterPusherServer(t.Server.GRPC, t.distributor)
 	}
 
+	// If the querier module is not part of this process we need to check if multi-tenant queries are enabled.
+	// If the querier module is part of this process the querier module will configure everything.
+	if !t.Cfg.isModuleEnabled(Querier) && t.Cfg.Querier.MultiTenantQueriesEnabled {
+		tenant.WithDefaultResolver(tenant.NewMultiResolver())
+	}
```

Comment:
I'm not really a fan of this. Ideally the multi-tenant resolver is configured globally based on the config. I actually assume that we are going to enable it by default. The only difference it makes is that `|` becomes an invalid character.

Reply:
I am wondering if we should do it for all the modules in a commonplace. What do you think?

Reply:
I'd like that. Where would that be?

Reply:
agreed this gets messy, `isModuleEnabled` is just doing a string contain, so we also have to make sure we're not running in `All` or `Read` as well if we take this approach.

Reply:
Maybe I misunderstood, what I meant to say is that the Querier is running in `Read` and `All`, but isn't detectable via the target string, so we might need to update this check to include those cases? If we took the approach of always registering this independent of whether the Querier is running or not it avoids all that messy checking.